zkevm-circuits/src/state_circuit/constraint_builder.rs

```diff
@@ -129,6 +133,16 @@ impl<F: Field> ConstraintBuilder<F> {
                 q.initial_value.clone() - q.initial_value_prev(),
             );
         });
+
+        self.add_lookup(
+            "if state_root changes, mpt_update exists in mpt circuit",
+            (
+                q.lexicographic_ordering_selector.clone()
+                    * (q.state_root() - q.state_root_prev())
```

Comment:
In my opinion, the lookup to mpt table should not be triggered by change of state_root, they should be triggered by some intrinsic rules (`Tag` is `AccountStorage` and it is the last row for a specified account / key, etc.). Following is the reasons:

1. the `state_root` col would be assigned arbitrarily in state circuit and there is no other way to provde them. For example we can assign identical value in `state_root` and all lookups are disabled.

2. Some entries are required in mpt table even it do not change state root. Consider a sole `SLOAD` to specified key or a `SELFBALANCE` op. We must provide it in mpt circuit but the state root after this op would never change.

Reply:
Thanks for catching this. I have changed the constraints to be what you suggested.

Reply:
Great. I think now it is ok.